src/standalone/notification/pythonExtensionRestartNotification.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable, inject } from 'inversify';
+import * as localize from '../../platform/common/utils/localize';
+import { IExtensionSyncActivationService } from '../../platform/activation/types';
+import { IPythonExtensionChecker } from '../../platform/api/types';
+import { IApplicationShell } from '../../platform/common/application/types';
+import { IDisposableRegistry } from '../../platform/common/types';
+import { noop } from '../../platform/common/utils/misc';
+import { IKernelProvider } from '../../kernels/types';
+
+// This class is responsible for watching if the Python Extension installation status changes, and if it does
+// update the users if any notebooks are running and need to be restarted.
+@injectable()
+export class PythonExtensionRestartNotification implements IExtensionSyncActivationService {
```

Comment:
Since the checker is in platform and our workflow here involved being able to look at kernels added the event to the checker to indicate installation and a new standalone service to look at the kernels and surface the message as needed.